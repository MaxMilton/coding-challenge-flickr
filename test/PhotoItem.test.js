'use strict';

const PhotoItem = require('../src/components/PhotoItem.html');

const data = {
  link: '11',
  media: {
    m: '22',
  },
  title: '33',
  author: '44',
  tags: '55',
};

describe('PhotoItem component', () => {
  it('throws error when missing prop data', () => {
    expect.assertions(1);
    const target = document.createElement('div');
    function wrapper() {
      new PhotoItem({ target });
    }
    expect(wrapper).toThrow();
  });

  it('renders correctly', () => {
    expect.assertions(2);
    const target = document.createElement('div');
    function wrapper() {
      new PhotoItem({ target, data });
    }
    expect(target.innerHTML).toMatchSnapshot();
    expect(wrapper).not.toThrow();
  });
});
