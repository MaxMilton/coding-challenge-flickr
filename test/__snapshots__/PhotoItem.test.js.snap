// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoItem component renders correctly 1`] = `""`;
